packages/catalog-generator/src/test/java/io/kaoto/camelcatalog/generators/CamelYAMLSchemaReaderTest.java

```diff
@@ -40,24 +42,32 @@ void shouldReturnJSONSchemaForEIP() {
         var eipSchema = camelYAMLSchemaReader.getJSONSchema(eipName);
 
         assertNotNull(eipSchema);
-        assertTrue(eipSchema.has("type"));
         assertTrue(eipSchema.has("properties"));
-        assertTrue(eipSchema.has("required"));
+        assertTrue(eipSchema.has("anyOf"));
     }
 
     @Test
     void shouldInlineDefinitions() {
-        var eipName = "resequence";
-        var eipSchema = camelYAMLSchemaReader.getJSONSchema(eipName);
+        var eipSchemaForResequence = camelYAMLSchemaReader.getJSONSchema("resequence");
 
-        assertTrue(eipSchema.has("items"));
-        var itemsNode = (ObjectNode) eipSchema.get("items");
-
-        assertTrue(itemsNode.has("definitions"));
-        var definitionsNode = (ObjectNode)  itemsNode.get("definitions");
+        assertTrue(eipSchemaForResequence.has("definitions"));
+        var definitionsNode = (ObjectNode)  eipSchemaForResequence.get("definitions");
 
         assertTrue(definitionsNode.has("org.apache.camel.model.config.BatchResequencerConfig"));
         assertTrue(definitionsNode.has("org.apache.camel.model.config.StreamResequencerConfig"));
         assertTrue(definitionsNode.has("org.apache.camel.model.language.ExpressionDefinition"));
+        assertTrue(definitionsNode.has("org.apache.camel.model.language.ConstantExpression"));
+        assertTrue(definitionsNode.has("org.apache.camel.model.PropertyDefinition"));
+
+        var eipSchemaForAggregate = camelYAMLSchemaReader.getJSONSchema("aggregate");
+
+        assertTrue(eipSchemaForAggregate.has("properties"));
+        var propertiesNode = (ObjectNode)  eipSchemaForAggregate.get("properties");
+
+        assertTrue(propertiesNode.has("optimisticLockRetryPolicy"));
+        var optimisticLockRetryPolicyNode = (ObjectNode)  propertiesNode.get("optimisticLockRetryPolicy");
+
+        assertTrue(optimisticLockRetryPolicyNode.has("$ref"));
+        assertTrue(optimisticLockRetryPolicyNode.get("$ref").asText().equalsIgnoreCase("#/definitions/org.apache.camel.model.OptimisticLockRetryPolicyDefinition"));
```

Comment:
Let's extract this test into a dedicated method with a descriptive name, in this case, it looks like you're checking whether the $ref has the right name and if it includes nested definitions